### ISSUE TITLE
Add Support for Google Tag Manager

### ISSUE THREAD

**Issue by rshokrizadeh** [zendesk/volunteer_portal#107]
Rather than supporting individual providers, we should add support for the developer to specify the container ID as a variable and it gets properly added to the snippet in the code base. This will allow people to add other tags into the app without needing to edit the code base directly.

**High in head:**

<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','INSERT CONTAINER ID HERE');</script>

**After the opening <body> tag:**

<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=INSERT CONTAINER ID HERE"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>

Will need to update the docs to explain where to specify the variable, similar to what we do currently for the Web Widget implementation.

**Comment by azaiszen** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Tagging events - Adding a volunteering type. Physical Volunteering vs Planning or Meetings etc.](https://trello.com/c/9JHmVHMK/6-tagging-events-adding-a-volunteering-type-physical-volunteering-vs-planning-or-meetings-etc)

**Comment by samzx** [comment]
resolved by #323

**Comment by rdzudzar** [comment]
Hi, I have a question about positioning Zendesk chat via GTM.

I have Zendesk Chat Widget GTM pixel code on clients website. And I found online that one can change position with the code below (from [developer zendesk](https://developer.zendesk.com/api-reference/widget/settings/?_ga=2.165456831.1600603858.1638319047-1290681618.1638319047#position)).
However, When I add custom html tag with this code - Zendesk chat does not change position at all. 
In addition, on my Zendesk dashboard: widget appearance has only option bottom right/left.

```
<script type="text/javascript">
  window.zESettings = {
    webWidget: {
      position: { horizontal: 'left', vertical: 'top' }
    }
  };
</script>

```

Thanks,
Robert
